### PR TITLE
Refactor iptables

### PR DESCRIPTION
**Pull request by baude**
Switch to a model where we use a rule generator approach for setup and
teardown.  This eliminates a significant amount duplicate code.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: src/commands/teardown.rs [hunk 1]
```diff
@@ -104,23 +106,30 @@ impl Teardown {
                                         "no network address provided",
                                     )
                                 })?;
-                                let td = TeardownPortForward {
-                                    network: network.clone(),
+                                let spf = SetupPortForward {
```
Comment (mheon) [inline src/commands/teardown.rs:109]: If we're refactoring already, maybe rename this so SetupPortForward isn't used to tear down port forwarding. `PortForwardConfig` maybe?

File: test/helpers.bash [hunk 2]
```diff
@@ -484,7 +484,7 @@ EOF
     done
 
     ## FIXME cleanup is broken atm
```
Comment (Luap99) [inline test/helpers.bash:486]: Cab you also remove this comment here

File: src/commands/teardown.rs (hunk 1 above)
Comment (Luap99) [reply src/commands/teardown.rs:109]: +1

**Comment by baude** [comment]
Fixed up local test failures and resolved comments.

**Pull request merged**